libs/data/bbhnet/data/distributions.py

```diff
@@ -71,3 +71,26 @@ def __call__(self, N: int) -> torch.Tensor:
         if self.low is not None:
             x = torch.clip(x, self.low)
         return x
+
+
+class VolumeDistributedSNR:
+    """
+    Sample from the following distribution which represents
+    universal distribution of signal-to-noise ratios (SNRs)
+    from volume distributed sources in volume
+    .. math::
+
+       p(\rho) = 3*\rho_0^3 / \rho^4
+
+    where :math:`\rho_0` is a representative minimum SNR
+    considered for detection. See, for example,
+    `Schutz (2011) <https://arxiv.org/abs/1102.5421>`_.
+    """
+
+    def __init__(self, rho_0: float = 4) -> None:
```

Comment:
It might be worth doing a couple things here, curious to get your thoughts @EthanMarx , especially in light of your proposal to generalize this submodule to `ml4gw`:
1. Use a more descriptive kwarg for the minimum value of the distribution, e.g. `min_val` or `low`
2. Don't give this a default value: it's a sufficiently defining property of the distribution that we should make users be explicit about the value they want.
3. Add the power as an argument (I'm fine having this default to 4), and just raise to the `-1 / (pow - 1)` at call time. This allows this both to be more general, as well as allows us to pass a torch scalar to this value that we can update over time, in case users want to manipulate their distribution through the course of training (e.g. start this off at 4, then have it slowly increase over time to focus on lower SNR events)

Reply:
Agree on all these points. Power laws with varying indices will certainly have other use cases.

Reply:
easy generalization. will do.

Reply:
Last comment: looks great and I like the addition of a max value, but would it be sensible to have this default to `float("inf")`. I have to imagine that will cover >99% of use cases

Reply:
OK

Reply:
Actually, another case where I think this can be used is the mass distribution for BBHs. And there, there is a cut off around 100 solar masses. So thinking about that, do we still want a default? I just kept the index default at 2, since its the integer after 1, no good reason apart from that.

Reply:
I mean default of infinity as the upper limit

Reply:
I'm still inclined to say set the default upper limit at infinity for a couple reasons:
1. My impression is that most use cases won't have a sensible upper limit to set, and are happy to rely on the statistics to functionally disallow really high values. This won't always be the case, but that's why they're defaults and not hard-coded values. But maybe I'm thinking too generally and in GW there will more often than not be a sensible max to set.
2. If (1) is true even 51% of the time, I'm inclined to set the default just because of how verbose `float("inf")` is and that it's not necessarily a super well-known bit of Python syntax (I would bet quite a few people will unnecessarily `import numpy` to use `np.inf`).

Reply:
ok.